### PR TITLE
[Feat] add option to rebase stalled PRs

### LINKED ISSUE #54: Update stale PRs
By "stale" I mean PRs that are no longer up-to-date with the base branch. This would happen every time someone makes a commit to `master`, for example - so I'm not sure if many people at all would want so many automated tests to be enqueued (e.g. 5 open renovations and 10 commits to master per day would result in 50 extra pull requests).

Probably best to wait for #52 first, at least.

### PR DESCRIPTION
**Pull request by SebastienElet**
Added the option `rebaseStalePrs` (not `rebaseStalePRs` to avoid `--rebase-stale-p-rs`).

Once merged, this will close #54

### CODE REVIEW

File: lib/api/github.js [hunk 1]
```diff
@@ -186,6 +186,9 @@ async function getPr(prNo) {
   if (pr.additions * pr.deletions === 1) {
     pr.canRebase = true;
   }
+  if (pr.base.sha !== config.baseCommitSHA) {
+    pr.isStalled = true;
```
Comment (rarkins) [inline lib/api/github.js]: Rename to `pr.isStale`

File: lib/worker.js [hunk 2]
```diff
@@ -155,8 +155,12 @@ async function updateDependency(upgrade) {
     let parentBranch = branchName;
     logger.debug(`Checking if branch ${branchName} needs updating`);
     const pr = await api.getBranchPr(branchName);
-    if (pr && pr.isUnmergeable) {
-      logger.verbose(`Existing ${pr.displayNumber} is not mergeable`);
+    if (pr && (pr.isUnmergeable || (pr.isStalled && config.rebaseStalePRs))) {
+      if (pr.isUnmergeable) {
+        logger.verbose(`Existing ${pr.displayNumber} is not mergeable`);
+      } else if (pr.isStalled) {
```
Comment (rarkins) [inline lib/worker.js]: Rename to `pr.isStale`

File: lib/config/definitions.js [hunk 3]
```diff
@@ -65,6 +65,12 @@ const options = [
     type: 'boolean',
   },
   {
+    name: 'rebaseStalePrs',
+    description: 'Rebase stale PRs',
+    type: 'boolean',
+    default: false,
+  },
```
Comment (rarkins) [inline lib/config/definitions.js]: Please move this new option to be below the `recreateClosed` one.

**Review by rarkins — CHANGES_REQUESTED** [review]

**Comment by SebastienElet** [comment]
@rarkins thanks for the feedbacks, it's done

**Comment by rarkins** [comment]
@SebastienElet I forgot to add - you will need to run `npm run update-docs` again after changing the config definitions file

**Comment by SebastienElet** [comment]
@rarkins i amended my commit to update the docs :-)

**Comment by rarkins** [comment]
@SebastienElet In the version on GitHub, the documentation is still out of order in the `*.md` files. Please try `npm run update-docs` again and then commit/push your changes. Once you reorder the definitions file, the docs should be reordered too.

**Comment by SebastienElet** [comment]
I forgot to `npm run build` before running `./bin/update-docs.sh`.

**Review by rarkins — APPROVED** [review]

**Comment by rarkins** [comment]
@SebastienElet there exists an npm script for that - you can run `npm run update-docs` and it does both.

Anyway, now merged and thanks again

**Comment by rarkins** [comment]
Published to npm version 8.4.0

**Pull request merged**